qbraid_qir/qasm3/visitor.py

```diff
@@ -328,8 +389,10 @@ def _visit_generic_gate_operation(self, operation: qasm3_ast.QuantumGate) -> Non
         Returns:
             None
         """
-        # TODO: maybe needs to be extended for custom gates
-        self._visit_basic_gate_operation(operation)
+        if operation.name.name in self._external_gates_map:
+            self._visit_external_gate_operation(operation)
+        else:
+            self._visit_basic_gate_operation(operation)
 
```

Comment:
Given that we are allowing external gates in our programs, there should be some failure handling here. Eg. if a gate is not present in the external gate map AND is not one of the basic gates, we should raise an error here rather than wait for [qasm operation mapper](https://github.com/qBraid/qbraid-qir/pull/182/files#diff-830829a9a5e98716c33880ec0328d6df1eceef2e90ca1161a263bfa544df1893R318)'s exception

Reply:
Makes sense! I added such a check in `pyqasm` but forgot it here. I'll add this as soon as I find the time!

Reply:
Yea, I was thinking that `pyqasm` should allow a gate to be marked as external even when it does not have a definition in the program. On thinking about it more, this doesn't make sense as there would be no way to validate the gate call. Moreover, once the validation checks have passed from `pyqasm` side only then will we be allowed to convert to QIR and so it is alright not to have a check here. My bad!